### PR TITLE
Add React best practice: Return Object from custom hooks instead of Array

### PR DESCRIPTION
**Pull request by thisismydesign**

### CODE REVIEW

File: react/best-practices.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  return {
+    someData,
+    someOtherData,
+  };
+};
+```
+
+BAD
+
+This way when using the hook, the order of variables would matter.
```
Comment (gomorizsolt) [inline react/best-practices.md]: Yep, that's right. On the other hand, using the object-approach means we're dependent on the names, aren't we?

Comment (thisismydesign) [reply react/best-practices.md]: You can choose to rename those explicitly. I think this approach is way less error-prone.

Comment (gomorizsolt) [reply react/best-practices.md]: > You can choose to rename those explicitly.

You mean the example below, right?

```js
const [someData: foo, someOtherData: bar] = useData();
```

Comment (thisismydesign) [reply react/best-practices.md]: Not sure :) If this works, that's super easy, worst case you can re-assign separately.

Comment (gomorizsolt) [reply react/best-practices.md]: Sorry, that should be enclosed with `{}`.

Comment (gomorizsolt) [reply react/best-practices.md]: Yep, it indeed works.

https://codesandbox.io/s/custom-hooks-object-o8oxw

Comment (thisismydesign) [reply react/best-practices.md]: Thanks. Added some usage examples: https://github.com/c-hive/guides/pull/35/commits/0cbe04d4a250b6651af1f9c3ca1f5a1f0cbf6f81

**Pull request merged**